integration/app/todo.state.ts

```diff
@@ -1,10 +1,14 @@
 import { State, Action, StateContext, Selector } from '@ngxs/store';
 
 export class AddTodo {
+  static type = 'AddTodo';
```

Comment:
we can say it is good practice but I don't think we can enforce it

Reply:
I think all our examples should be
`static readonly type = 'section.subsection.actionname'`

I think we should promote this as a best practice so that it's easier to filter out actions from a specific section of your app, and also minimize collisions.

**Sidenote**
I use this quite a lot in my java backend apps, where I use a reverse namespace
`se.radley.myproject.user.settings.SaveSettings`

and I can then set the log level for a specific package thus get debug output for all my user settings stuff. (also runtime changeable in spring boot via the config rest endpoint)

Maybe we could implement this in the ngxs logger, so that we can have the logger enabled even at prod time, but easily via the chrome debugger console set the logging level for a certain path?

Reply:
ill update everything to use readonly